apps/frontend/src/components/WorkoutMenu/WorkoutListItem.tsx

```diff
@@ -81,3 +72,27 @@ export const WorkoutListItem = ({
     </Grid>
   );
 };
+
+const WorkoutIcon = ({ type }: { type: WorkoutType }) => {
+  if (type === 'library') {
+    return (
+      <Tooltip label={`Workout is from the workout library`} placement="left">
+        <Center>
+          <Icon as={Book} />
+        </Center>
+      </Tooltip>
+    );
+  }
+  return (
+    <Tooltip
+      label={`Workout is stored ${
+        type === 'remote' ? 'in the browser' : 'remotely'
+      }`}
+      placement="left"
+    >
+      <Center>
+        <Icon as={type === 'local' ? Hdd : Cloud} />
+      </Center>
+    </Tooltip>
+  );
```

Comment:
Nice! Could probably have been separated even more like this:

```suggestion
  const iconAndDescription = (type: WorkoutType) => {
    switch(type) {
      case "library":
        return {
          icon: Book,
          description: "Workout is from the workout library"
        }
      case "local":
        return {
          icon: Hdd,
          description: "Workout is stored in the browser"
        }
      case "remote":
        return {
          icon: Cloud,
          description: "Workout is stored remotely"
        }
    }
  };
  
  const { icon, description } = iconAndDescription(type);
  
  return (
    <Tooltip
      label={description}
      placement="left"
    >
      <Center>
        <Icon as={icon} />
      </Center>
    </Tooltip>
  );
```

No idea how close this is to compiling and working, but you get the idea! 🤠

Reply:
This is of course compiling and working 🪨 🚀